src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -119,10 +119,13 @@ else if (!savedSenderID.equals(senderID)) {
                         editor.commit();
                     }
 
-                    if (gCachedExtras != null) {
+                    if (!gCachedExtras.isEmpty()) {
                         Log.v(LOG_TAG, "sending cached extras");
-                        sendExtras(gCachedExtras);
-                        gCachedExtras = null;
+                        Iterator<Bundle> gCachedExtrasIterator = gCachedExtras.iterator();
```

Comment:
You are creating an Iterator here but there is a possible concurrency issue if a new push notification comes in while you are iterating through the current notifications. Probably best off to make a copy of `gCachedExtrasIterator` then empty `gCachedExtrasIterator` before you start iterating.